packages/neon/neon/lib/src/models/account.dart

```diff
@@ -2,20 +2,34 @@ import 'dart:convert';
 
 import 'package:collection/collection.dart';
 import 'package:crypto/crypto.dart';
-import 'package:flutter/foundation.dart';
 import 'package:json_annotation/json_annotation.dart';
+import 'package:meta/meta.dart';
 import 'package:nextcloud/nextcloud.dart';
 
 part 'account.g.dart';
 
+/// Credentials interface
+@internal
+@immutable
+abstract interface class Credentials {
```

Comment:
Ideally we'd use this interface everywhere we have a `withCredentials` constructor but sadly go_router wants the base types themselves for the routes. Meaning we'd do something like Credentials -> basetypes -> Credentials wich is way more boilerplate than it already needs to be.

Reply:
Is this anything that can be fixed in this PR?

Reply:
It can but I don't like all the boilerplate.
Especially with the redirects it get's worse:

Credential -> string -> credential  -> string -> Credential -> basetypes.

With the three steps being Login*Route, AddAccount*Route and *Page.